akka-actor-typed/src/main/scala/akka/actor/typed/delivery/internal/ProducerControllerImpl.scala

```diff
@@ -486,29 +545,92 @@ private class ProducerControllerImpl[A: ClassTag](
       s.copy(confirmedSeqNr = newMaxConfirmedSeqNr, replyAfterStore = newReplyAfterStore, unconfirmed = newUnconfirmed)
     }
 
-    def receiveStoreMessageSentCompleted(seqNr: SeqNr, m: A, ack: Boolean) = {
-      if (seqNr != s.currentSeqNr)
-        throw new IllegalStateException(s"currentSeqNr [${s.currentSeqNr}] not matching stored seqNr [$seqNr]")
+    def receiveStoreMessageSentCompleted(seqNr: SeqNr): Behavior[InternalCommand] = {
+      if (seqNr == s.storeMessageSentInProgress) {
```

Comment:
I think it can happen if there is a timeout of the ask to the durableStore. Then there is a retry. Maybe that could trigger a "duplicate" reply (for wrong seqNr) later. I'll add a debug log, but I think it's correct to filter such duplicates.